### PR TITLE
Bump to new FLINT_jll

### PR DESCRIPTION
**Pull request by thofma**

### CODE REVIEW

**Comment by wbhart** [comment]
There will be a 2.8.4. The same bug we fixed in 2.8.3 exists in
fmpz_mod_poly.

We will also have the ARM patch in 2.8.4 and we are going to remove the C++
documentation, since it is incompatible with different versions of Sphinx
in different ways.

This will occur next week of course.

Bill.

File: Project.toml [hunk 1]
```diff
@@ -13,7 +13,7 @@ Pkg = "44cfe95a-1eb2-52ea-b672-e2afdf69b78f"
 
 [compat]
 BinaryProvider = "0.4, 0.5"
-FLINT_jll = "~200.800.100"
+FLINT_jll = "~200.800.300"
```
Comment (thofma) [inline Project.toml:16]:
```suggestion
FLINT_jll = "^200.800.300"
```

Comment (fingolfin) [reply Project.toml]: Really? I thought we deliberately wanted to be conservative here? i mean, not that I mind, I don't care about older Julia versions, but...

Comment (thofma) [reply Project.toml]: `~200.800.300` excludes `~200.800.301`.

Comment (thofma) [reply Project.toml]: Making it impossible to load this with `FLINT_jll` on `>= 1.6`.

Comment (thofma) [reply Project.toml]: Or maybe I did something wrong, let me check again.

File: Project.toml [hunk 2]
```diff
@@ -13,7 +13,7 @@ Pkg = "44cfe95a-1eb2-52ea-b672-e2afdf69b78f"
 
 [compat]
 BinaryProvider = "0.4, 0.5"
-FLINT_jll = "~200.800.100"
+FLINT_jll = "^200.800.300"
```
Comment (thofma) [inline Project.toml:16]:
```suggestion
FLINT_jll = "~200.800.300"
```

File: Project.toml (hunk 1 above)
Comment (thofma) [reply Project.toml:16]: My bad. I changed it back. Should be good to go.

**Pull request merged**